### PR TITLE
[CI] Run R check in GitHub Actions

### LINKED ISSUE #6068: [CI] Run R check in GitHub Actions
Currently, GitHub Actions run the unit tests for R, but does not run `R CMD CHECK --as-cran xgboost*.tar.gz`. This check is essential for uploading XGBoost to CRAN.

### PR DESCRIPTION
**Pull request by ahmedkrmn**
Closes #6068

### CODE REVIEW

**Comment by ahmedkrmn** [comment]
Hi @hcho3,
I tried several approaches, but all of them failed.
First, doing as such:
```yml
run: make Rcheck
```
The build failed because the `float` package wasn't installed, so I added it. 
Then It failed again with the error: "pdflatex is not available".

Doing so:
```yml
run: |
  make Rbuild
  R CMD check --as-cran xgboost*.tar.gz --no-manual
```
Where I added the `--no-manual` flag to skip pdf file generation also fails!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -310,3 +310,8 @@ jobs:
     - name: Test R
       run: |
         python tests/ci_build/test_r_package.py --compiler="${{ matrix.config.compiler }}" --build-tool="${{ matrix.config.build }}"
+
+    - name: Check R
+      run: |
+        make Rbuild
+        R CMD check --as-cran xgboost*.tar.gz --no-manual
```
Comment (hcho3) [inline .github/workflows/main.yml:317]: @ahmedkrmn Can you move this to a new job? The `test-with-R` job is running on Windows, and it's harder to get the R check work on Windows. Make sure to set `os` to Ubuntu Linux.

**Comment by hcho3** [comment]
@ahmedkrmn Any updates? Would you like some help?

**Comment by ahmedkrmn** [comment]
Hi Philip,
I was busy lately, will have another look this weekend.

**Comment by ahmedkrmn** [comment]
> @ahmedkrmn Any updates? Would you like some help?

Hi Philip, thanks for offering help!
I'm pretty busy now so please feel free to re-assign this issue if someone wants to work on it.

**Comment by hcho3** [comment]
Sure, let me take over.

**Pull request closed without merging**